cruise-control-client/cruisecontrolclient/client/cccli.py

```diff
@@ -5,134 +5,24 @@
 
 # To be able to easily parse command-line arguments
 import argparse
+from typing import Any, Dict, Optional, Set, Tuple, Type
 
 # To be able to easily pass around the available endpoints and parameters
-from cruisecontrolclient.client.ExecutionContext import ExecutionContext
-
+from cruisecontrolclient.client.ExecutionContext import (AVAILABLE_ENDPOINTS,
+                                                         NAME_TO_ENDPOINT,
+                                                         NON_PARAMETER_FLAGS,
+                                                         FLAG_TO_PARAMETER_NAME)
 # To be able to instantiate Endpoint objects
 import cruisecontrolclient.client.Endpoint as Endpoint
 
+# To be able to bundle the parameters sent to an endpoint
+from cruisecontrolclient.client.ParameterSet import ParameterSet
+
 # To be able to make long-running requests to cruise-control
 from cruisecontrolclient.client.Responder import CruiseControlResponder
 
 
-def get_endpoint(args: argparse.Namespace,
```

Comment:
This method contains much more than 'getting the endpoint', so I've partitioned it into more targeted methods that further allow the decoupling of the argparse.Namespace class being used to carry all the parameters.